IntegrationTests/test_runner.py

```diff
@@ -0,0 +1,352 @@
+"""
+This is the SOM integration test runner file. Pytest automatically discovers
+this file and will find all .som test files in the below directories.
+"""
+
+import subprocess
+from pathlib import Path
+import os
+import sys
+import pytest
+import yaml
+import conftest as external_vars
+
+
+def debug(message):
+    """
+    Take a string as a mesasage and output if DEBUG is true
+    """
+    if external_vars.DEBUG is True:
+        print(message)
+
+
+def debug_list(message_list, prefix="", postfix=""):
+    """
+    Take a list of messages and output if DEBUG is true, with a prefix and a postfix
+    """
+    if external_vars.DEBUG is True:
+        for message in message_list:
+            print(prefix + str(message) + postfix)
+
+
+def locate_tests(path, test_files):
+    """
+    Locate all test files that exist in the given directory
+    Ignore any tests which are in the ignoredTests directory
+    Return a list of paths to the test files
+    """
+    # To ID a file will be opened and at the top there should be a comment which starts with VM:
+    for file_path in Path(path).glob("*.som"):
+        # Check if the file is in the ignored tests (Check via path, multiple tests named test.som)
+        with open(file_path, "r", encoding="utf-8") as f:
+            contents = f.read()
+            if "VM" in contents:
+                test_files.append(file_path)
+
+    return test_files
+
+
+def read_directory(path, test_files):
+    """
+    Recursively read all sub directories
+    Path is the directory we are currently in
+    testFiles is the list of test files we are building up
+    """
+    for directory in Path(path).iterdir():
+        if directory.is_dir():
+            read_directory(directory, test_files)
+        else:
+            continue
+
+    locate_tests(path, test_files)
+
+
+def assemble_test_dictionary(test_files):
+    """
+    Assemble a dictionary of
+    name: the name of the test file
+    stdout/stderr: the expected output of the test
+    """
+    tests = []
+    for file_path in test_files:
+        test_dict = parse_test_file(file_path)
+        if test_dict is None:
+            continue
+        tests.append(test_dict)
+
+    return tests
+
+
+def parse_test_file(test_file):
+    """
+    parse the test file to extract the important information
+    """
+    test_info_dict = {
+        "name": test_file,
+        "stdout": [],
+        "stderr": [],
+        "customCP": "NaN",
```

Comment:
This doesn't match the readme:

```suggestion
        "custom_classpath": None,
```

But why is this `NaN`? Couldn't this be something more conventional like `None`?